### ISSUE TITLE
[Bug]: No intrusion alarm notification

### ISSUE THREAD

**Issue by Mense1978** [bvis/aegis-hass#372]
### Describe the bug

I have 5 zones configured in my Ajax alarm system. When any of the zones is triggered, no notification is received in Home Assistant.

I tried capturing events, but it only detects when a zone is armed or disarmed.

### Steps to reproduce

I arm any zone, either from the Aegis panel or the Ajax app, and then trigger the zone by simulating an intrusion.

I check the logs and try to capture the event, but no information is reported.

### Expected behavior

There should be an event indicating that an intrusion has occurred.

### Home Assistant version

2026.7.4

### Integration version

Other / not listed

### Installation method

HACS

### App label (co-branded app)

aJAX

### FCM push notifications configured?

Yes

### Relevant logs

```text

```

**Comment by bvis** [comment]
Thanks for the report. Before anything else, one reassurance and one correction of expectations, because they change what we're looking for.

**Alarm and zone state do not depend on push at all.** If your groups arm and disarm correctly in Home Assistant, the part that matters is working — that state comes from polling and from the hub connection, never from a notification. What push adds is low-latency *events*: the alarm firing, a doorbell ring, and similar. So what appears to be missing here is the event, not the state.

**Which version are you on?** You selected "Other / not listed", and that matters a lot right now: builds `1.15.1-beta.6` through `beta.10` shipped a packaging defect that broke an unrelated read on every install, and there is a separate bug (#373) that can stop push permanently. Please tell me the exact version from HACS, and if it isn't the newest beta, update to **`1.15.1-beta.11`** before collecting anything below.

Then I need debug logs, because "no information is reported" can mean three completely different things and they need opposite fixes from me. Add this to `configuration.yaml`, restart, trigger a real intrusion on an armed group, and then look at the log:

```yaml
logger:
  default: warning
  logs:
    custom_components.aegis_ajax: debug
    firebase_messaging: debug
```

Grep for these three, and tell me which you see:

1. **`Push notification received`** — logged for every push that arrives, whatever its content.
2. **`Push event parsed`** — logged only after a push arrived *and* was decoded.
3. **`Incorrect padding`** or **`shutting down FcmPushClient`**.

The combination tells us where the problem is:

- **Neither 1 nor 2, and no 3** → nothing is reaching Home Assistant. That is almost always the Ajax account's permission level: an account that is not administrator / full access receives no events at all, even with every notification category enabled in the app. This is what #234 and #319 turned out to be, and it is documented in the README under push notifications. Since you say arm/disarm *is* being detected, this is less likely in your case — which is itself informative.
- **1 present, 2 missing** → the push arrives and the integration fails to decode it. That one is mine, and I'd want the payload.
- **3 present** → this is #373: one message the push library cannot decrypt shuts the connection down, and because it is never acknowledged it gets redelivered and kills it again on every reconnect. From the outside this looks exactly like the first case, which is why I'm asking you to grep for it explicitly. A fix is in review right now.

One more question that would narrow this a lot on its own, and costs you nothing but a note: when you trigger the intrusion, **does the Ajax app on your phone show the alarm?** If the app gets it and Home Assistant doesn't, we are looking at delivery to this specific account. If the app doesn't either, the hub is not reporting the intrusion the way we assume and that is a different investigation.

Finally, so I can check the right event path: which entity are you watching for the intrusion — the alarm panel going to `triggered`, a specific sensor's binary sensor, or an `aegis_ajax_event`-type automation trigger? Pasting the automation or trigger you tried would help.

**Comment by bvis** [comment]
Update while you gather the logs: **`1.15.1-beta.12`** is out, and it fixes a bug (#373) that can stop push notifications permanently and silently. That is one of the three possibilities I listed, so please install this first — it may remove one branch of the investigation before you spend time on it.

HACS → the integration → three-dot menu → Redownload → tick "Show beta versions" → pick `1.15.1-beta.12` → restart Home Assistant.

Then trigger the intrusion again and collect the log as described above. The questions that still matter most, whatever the logs say:

1. Does the **Ajax app on your phone** show the alarm when you trigger it?
2. Which entity or trigger are you watching for the intrusion in Home Assistant?

Those two answers narrow this down more than the logs do on their own.

**Comment by bvis** [comment]
Update: **`1.15.1` is now out as a stable release**, so you no longer need to enable "Show beta versions" — HACS will offer it as a regular update. It includes the push fix mentioned above (#373).

After updating and restarting, the plan is unchanged: trigger a real intrusion on an armed group with debug logging on, and grep for the three markers listed above. And the two questions that narrow this the most still stand:

1. Does the **Ajax app on your phone** show the alarm when you trigger it?
2. Which entity or trigger are you watching for the intrusion in Home Assistant?

**Comment by Mense1978** [comment]
Hello, I aswer in order.
1.The app shows the alarms.
I am trying to get the event, using developer tools/event[image: imagen.png]

**Comment by bvis** [comment]
Thank you — and your first answer already rules out the most likely cause. **The Ajax app shows the alarm**, so the events are reaching your account. That eliminates the account-permission branch (#234 / #319), which was my leading theory and is the one that would have been hardest to fix. Good news.

Your second answer did not survive the trip: you replied by email, and the screenshot came through as the text `[image: imagen.png]` with no file attached. GitHub drops inline images from email replies. Could you post it again **from the GitHub web page** (or drag the file into the comment box) rather than by email?

What I need to see from it is specific, so if the screenshot is inconvenient, plain text answers work just as well:

1. In **Developer tools → Events**, what did you type into "Listen to events"? To catch these you need exactly `aegis_ajax_event` — the integration fires that one event type, and listening to `*` will bury it in hundreds of unrelated events.
2. With that listener running, trigger the intrusion. **Does anything appear at all?** If yes, paste the whole event block — the `event_type` and `raw_tag` fields tell me which path fired.
3. If nothing appears, that is the answer I need most, and it points at the push path rather than the event mapping.

And whatever the events window shows, the log grep from earlier still matters, because it separates "nothing arrived" from "something arrived and was not understood". With debug logging on, trigger the intrusion and search the log for `Push notification received` and `Push event parsed`. Which of the two appears — and whether either does — narrows this faster than anything else.

**Comment by Mense1978** [comment]
Hello, i tryied everything with no luck.
Nothing shows on events nor registry.

Thanks

**Comment by bvis** [comment]
@Mense1978 — your screenshots are not reaching me. This has now happened twice: replying by email strips the attached image and GitHub shows only the text `[image: imagen.png]`. Nothing you did wrong, and I should have made that clearer the first time.

**So please do not send a screenshot at all.** Everything I need can be typed as text, and two short answers would move this further than any image:

1. In **Developer tools → Events**, type `aegis_ajax_event` into "Listen to events" and press Start. Then trigger the intrusion. **Does anything appear?** Just "yes" or "no" is enough. If yes, copy the text of the block that appears and paste it.
2. Turn on debug logging, trigger the intrusion, and tell me which of these two lines you find in the log — **either, both, or neither**:
   - `Push notification received`
   - `Push event parsed`

That second answer is the important one. We now know from another user's capture that the arm/disarm detection you already have working comes over a *different* channel than the alarm notification, so the fact that arm/disarm works tells us nothing about why the alarm does not. Those two log lines are what separates "nothing is arriving" from "something arrived and was not understood" — and they point at completely different causes.

If it is easier, you can reply from the GitHub website rather than by email, which also avoids the image problem.

**Comment by Mense1978** [comment]
Hello again.

I answer in order.
1. Nothing happens
2. I paste the info gathered on registry. Activated zone name is : Casa
David Perimetral

Thank you for your help

**Comment by bvis** [comment]
@Mense1978 — your "nothing happens" is a real result, not a dead end. Thank you for running it.

**One question this time, and the answer is one word.** No screenshots, no logs, no pasting — those keep getting lost on the way here and that is not your fault.

Do exactly what you did before, but **arm or disarm** instead of triggering the intrusion:

1. Developer tools → Events → type `aegis_ajax_event` → Start.
2. Arm or disarm from the Ajax app.
3. **Does anything appear? Yes or no.**

That is the whole question. You can reply with just "yes" or "no".

**Why this one answer decides it.** In Home Assistant an intrusion alarm can only arrive as a *notification* — there is no second route for it. Arming is different: the panel showing "armed" comes from a separate channel that does not use notifications at all, which is why arming can look like it works while notifications are completely dead. So "arming works" has never told us anything, and I should have realised that sooner instead of asking you for more logs.

The test above closes that gap, because an arm produces an **event** only through the same notification path the alarm needs:

- **Yes, something appears** → notifications are working, and only the alarm event is missing. That is a much smaller problem and it is on my side.
- **No, nothing appears** → notifications are not being delivered to Home Assistant at all. That explains the alarm completely, and we would then work on the notification setup instead of the alarm.

Either answer moves this forward. And your earlier answer already ruled out the worst case: the Ajax app on your phone shows the alarm, so your account does receive it — nothing is wrong with your Ajax system or your zones.

One practical note: replying **from the GitHub website** rather than by email avoids the problem that ate your last two answers. Email strips attachments and part of the text before it reaches me.

**Comment by Mense1978** [comment]
Hi, I answer in order.

1 and 2 Nothing appears on event.

I put ther debug file

Thank you for your help

**Comment by bvis** [comment]
@Mense1978 — that is the answer that settles it. Thank you for running it.

**Nothing appears on arm or disarm either**, which means push notifications are not reaching Home Assistant at all — not just the alarm one. An arm produces an event only through the same channel the alarm needs, so with both silent the problem is the channel itself. Your earlier observation that arming "was detected" was the alarm panel changing state, and that travels a different route which does not use notifications at all. That is why it kept working, and why it kept misleading us both.

One practical note: the debug file did not arrive again — email strips attachments before they reach me. Please do not send it. You do not need to.

**One thing to do, and the answer is one word.**

There is a known fault that produces exactly this. A single push message the integration could not read shut the connection down permanently, and because that message stays waiting on Ajax's server, every reconnection died on the same one. Restarting Home Assistant did not help, and neither did rebooting. It was fixed in version **1.15.1**. Your report lists the integration version as *Other / not listed*, so I cannot tell whether your build has that fix — and until I know, nothing else is worth testing.

1. In HACS, update Aegis Ajax to the newest available version.
2. Restart Home Assistant.
3. Repeat exactly the test you just did: Developer tools → Events → `aegis_ajax_event` → Start, then arm or disarm from the Ajax app.
4. **Does anything appear now? Yes or no.**

Either answer moves this forward. **Yes** means that was the fault, and your intrusion alarm should now arrive too. **No** rules it out, and the next thing I would look at is your Ajax account's notification permissions — the other thing that can silence this completely, and something we have seen before on accounts with a limited role.

Replying **from the GitHub website** rather than by email will keep your answer intact.

**Comment by Mense1978** [comment]
Hi, I only get this:

event_type: state_changed
data:
  entity_id: alarm_control_panel.orbera_16_exterior
  old_state:
    entity_id: alarm_control_panel.orbera_16_exterior
    state: disarmed
    attributes:
      code_format: null
      changed_by: null
      code_arm_required: false
      hub_id: 00145D7A
      space_id: 65846d85d75147184b004b8c
      group_id: '00000003'
      group_name: Exterior
      connection_status: ONLINE
      friendly_name: Orbera 16 Exterior
      supported_features: 2
    last_changed: '2026-08-06T14:46:27.099050+00:00'
    last_reported: '2026-08-06T14:46:27.099050+00:00'
    last_updated: '2026-08-06T14:46:27.099050+00:00'
    context:
      id: 01KZBRMDGV4MYAXYYJ9D7KNPW1
      parent_id: null
      user_id: null
  new_state:
    entity_id: alarm_control_panel.orbera_16_exterior
    state: armed_away
    attributes:
      code_format: null
      changed_by: null
      code_arm_required: false
      hub_id: 00145D7A
      space_id: 65846d85d75147184b004b8c
      group_id: '00000003'
      group_name: Exterior
      connection_status: ONLINE
      friendly_name: Orbera 16 Exterior
      supported_features: 2
    last_changed: '2026-08-06T14:47:21.523492+00:00'
    last_reported: '2026-08-06T14:47:21.523492+00:00'
    last_updated: '2026-08-06T14:47:21.523492+00:00'
    context:
      id: 01KZBRP2NKXZ2HGKEEWW7CXBQF
      parent_id: null
      user_id: null
origin: LOCAL
time_fired: '2026-08-06T14:47:21.523492+00:00'
context:
  id: 01KZBRP2NKXZ2HGKEEWW7CXBQF
  parent_id: null
  user_id: null


thanks

**Comment by bvis** [comment]
@Mense1978 — thank you, and that paste tells me something useful even though it is not what I asked for. **No more logs.** You have done four rounds of these and I have not managed to make them pay off, so we are stopping that.

What you sent shows your panel going from *disarmed* to *armed_away*. That part of the integration is working perfectly. But it arrives by a different route than notifications do, so it confirms what we already knew and cannot tell us about the alarm. The problem is that **notifications are not arriving at all** — and rather than measure that again, let us just try to fix it.

## What to do, and nothing else

Notifications need a registration between your Home Assistant and Ajax. That registration can break in a way that never repairs itself, and the fix is to make a new one. Five steps.

**1. Update first.**
Go to **HACS → Aegis for Ajax → the three dots → Redownload**, pick the newest version offered, then restart Home Assistant. There was a fault that killed notifications permanently and it was repaired in a later version, so this alone may be all that is needed.

**2. Note down your four Ajax notification values.**
Go to **Settings → Devices & Services → Aegis for Ajax → Configure**. You will see four boxes with long values in them. Copy all four somewhere safe — a note on your phone is fine. **Do not post them here**, they are private to your account.

**3. Empty those four boxes and save.**
Delete the contents of all four, then submit. This removes the broken registration.

**4. Restart Home Assistant.**

**5. Put the four values back and save, then restart once more.**
This makes Ajax issue a completely new registration.

## Then the only test, and it is short

Arm or disarm from the Ajax app on your phone. Then look at your alarm panel in Home Assistant and, separately, at **Settings → Devices & Services → Aegis for Ajax**, where a notification problem is reported as a warning if there is one.

**Tell me one of two things:**

- **"There is a warning"** — and copy the text of it. That names the cause directly.
- **"No warning"** — then trigger the intrusion once more and tell me whether the alarm appears in Home Assistant this time. Just *yes* or *no*.

## Two things so you are not left wondering

**Your Ajax system is fine.** Your zones, your sensors and your hub are all working, and your phone receives the alarm — that has been true since your first message. What is broken is only the delivery of notifications to Home Assistant, which is our side.

**If step 5 does not fix it**, the next thing I will look at is whether the Ajax account you used has full permissions, because an account with a restricted role receives no notifications and produces no error at all — which would match everything you have seen. That needs no work from you, just an answer about how the account was created.

One small practical note: if you can, **answer from the GitHub website** rather than by replying to the email. Email has removed your screenshots twice and part of your text once, and none of that was your fault.

**Comment by Mense1978** [comment]
Hi I have installed 1.16.0 beta 7 and nothing happened.

Thanks

**Comment by bvis** [comment]
@Mense1978 — thank you for updating, and I am not asking for logs again. But the update was **step 1 of five**, and it is the least important of them. Steps 2 to 5 are the actual repair: they throw away the broken notification registration and make Ajax issue a brand new one. Installing a newer version does not do that on its own, so "nothing happened" is the expected result of step 1 alone.

Please do the four remaining steps. Everything happens inside Home Assistant, nothing on your Ajax system is touched, and it takes about ten minutes:

2. Go to **Settings → Devices & Services → Aegis for Ajax → Configure**. Copy the four long values somewhere safe — a note on your phone is fine. **Do not post them here**, they are private to your account.
3. Delete the contents of all four boxes and submit.
4. Restart Home Assistant.
5. Put the four values back, submit, and restart once more.

Then arm or disarm from the Ajax app on your phone, and trigger the intrusion alarm once.

**Two questions, and one word answers each:**

1. After those steps, does the alarm appear in Home Assistant? **Yes** or **no**.
2. The Ajax account that Home Assistant uses — is it the **same** account you use in the Ajax app on your phone, or a **separate** one you were invited to? An account with a limited role receives no notifications at all and reports no error, which would match everything you have seen so far.

I also want to be straight with you about where this stands. I have asked for measurements four times and not one has reached me — your email replies strip the attachments and the images before they arrive, so I have never seen a single line from your log, and everything I know about your install is what you have typed. That is not your fault. It does mean these five steps are the last thing I can offer from here. **If they cannot be done, I will close this issue**, because anything further would be guessing in public. Closing it would not be a judgement on the report, and I will reopen it the moment there is something concrete to work from.

Your Ajax system itself is fine throughout all of this — the hub, the zones and the sensors work, and your phone receives the alarm. What is broken is only the delivery of notifications into Home Assistant, which is our side.

**Comment by bvis** [comment]
@Mense1978 — a gentle nudge, no rush and no logs needed. Everything from the last message still stands: the only thing pending is steps 2 to 5 (the ten-minute reset of the notification registration) and the two one-word questions — did the alarm appear afterwards, and is the Home Assistant account the same one as on your phone or a separate invited one?

If I have not heard back by **August 23** I will close this issue as announced — not as a judgement on the report, just to keep the tracker honest. It reopens the moment you can run those steps and tell me the two answers, whenever that is. Enjoy the summer.

**Comment by bvis** [comment]
Closing for now — not because the report is wrong, but because it has been unconfirmable for two rounds: the remaining test steps (2–5 from the message above) have been pending since 6 Aug, and without them the investigation cannot tell an Ajax-side delivery gap from an integration defect on this install.

**This closes the ticket, not the question.** Two things keep it alive elsewhere:

- #426 is now measuring the same territory (whether intrusion alarm events reach Home Assistant at all) on an install whose reporter can run the tests, and the alarm panel is gaining a derived `triggered` state from exactly that event.
- **Reopen condition:** if you run the arm/disarm listen test (Developer Tools → Events → `aegis_ajax_event`) and events appear there but a real intrusion alarm produces none, comment here and this reopens immediately — that datum alone discriminates, no logs needed.

Thank you for the rounds you did run.
